runtime/src/snapshot_utils.rs

```diff
@@ -932,46 +1551,86 @@ fn verify_snapshot_version_and_folder(
 }
 
 #[allow(clippy::too_many_arguments)]
-fn rebuild_bank_from_snapshots(
-    snapshot_version: &str,
+fn rebuild_bank_from_snapshots<P, Q>(
+    full_snapshot_version: &str,
+    incremental_snapshot_version: Option<&str>,
     frozen_account_pubkeys: &[Pubkey],
-    unpacked_snapshots_dir: &Path,
+    full_snapshot_unpacked_snapshots_dir: P,
+    incremental_snapshot_unpacked_snapshots_dir: Option<Q>,
```

Comment:
Might be a good idea to fold `incremental_snapshot_unpacked_snapshots_dir` and `incremental_snapshot_version` into a tuple under the same `Option` here since they are both Some or None, maybe a struct `UnpackedSnapshotDir`? (maybe could use this for the `full_snapshot_unpacked_snapshots_dir + full_snapshot_version` as well

Reply:
I like it. Will do!

Reply:
Done!